test/integration/version_upgrade_test.go

```diff
@@ -82,11 +82,13 @@ func TestRunningBinaryUpgrade(t *testing.T) {
 		if arm64Platform() {
 			// arm64 KIC driver is supported starting from v1.17.0
 			legacyVersion = "v1.17.0"
-		} else {
-			// v1.8.0 would be selected, but: https://github.com/kubernetes/minikube/issues/8740
-			legacyVersion = "v1.9.0"
 		}
 	}
+	// the version containerd in ISO was upgraded to 1.4.2
+	// we need it to use runc.v2 plugin
+	if ContainerRuntime() == "containerd" {
```

Comment:
this seems to be setting it globally for both KIC and VM drivers
previous logic has a different legacy per driver


since this is getting a complex, consider making a data structure to simplify it, for example how about  map. the map would look something like this
```
legacayMap = map[string][string] semver

legacayMap["KIC"] ["containerd"] = v1.X.X
legacayMap["VM"] ["Docker"] = v1.X.X
```
and that could be a constant in our constants package and be easily manintained

and then you could use a helper func to see if a driver is KIC or NOT

Reply:
Checked this test for [driver=docker, runtime=containerd] config. It also fails if legacy version is less than 1.16.0. It looks like for conainerd we have to use the same legacy for both kvm and kicbase